### PR TITLE
Fix unit tests

### PR DESCRIPTION
**Pull request by paragonie-scott**

### CODE REVIEW

File: src/File.php [hunk 1]
```diff
@@ -771,10 +785,12 @@ private static function getFileVersionConfigFromMajorMinor($major, $minor)
      */
     final public static function readBytes($stream, $num)
     {
-        if ($num <= 0) {
+        if ($num < 0) {
```
Comment (defuse) [inline src/File.php]: Good catch! It's definitely valid to try and read 0 bytes.

File: src/File.php [hunk 2]
```diff
@@ -400,6 +408,12 @@ public static function decryptResource($inputHandle, $outputHandle, Key $key)
                 'Output handle must be a resource!'
             );
         }
+        $stat = \fstat($inputHandle);
```
Comment (defuse) [inline src/File.php]: I guess we can't (or it's unreasonably complex) do this in a TOCTTOU-free way.

File: src/Exception/InvalidInput.php [hunk 3]
```diff
@@ -0,0 +1,7 @@
+<?php
+namespace Defuse\Crypto\Exception;
+
+class InvalidInput extends \Defuse\Crypto\Exception\CryptoException
```
Comment (defuse) [inline src/Exception/InvalidInput.php]: This PR adds this exception, but where does it get used?

**Comment by defuse** [comment]
Will merge once the `InvalidInput` question is answered :)

File: src/Exception/InvalidInput.php (hunk 3 above)
Comment (paragonie-scott) [inline src/Exception/InvalidInput.php]: https://github.com/paragonie/php-encryption/blob/fix-taylors-dec12-branch/src/File.php#L407

It looks like there was already code written for these exceptions without defining it.

**Comment by defuse** [comment]
Thank you!!

**Pull request merged**